controller/finalizer.go

```diff
@@ -87,68 +87,86 @@ func (f *Controller) addFinalizer(obj interface{}) (bool, error) {
 // was set by operatorkit. The removal of a finalizer will be retried and a fresh
 // object will get fetched from k8s if the ResourceVersion is out of date.
 func (c *Controller) removeFinalizer(ctx context.Context, obj interface{}) error {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
 	finalizerName := getFinalizerName(c.name)
+	selfLink := accessor.GetSelfLink()
 
-	c.logger.Log("level", "debug", "message", fmt.Sprintf("removing finalizer '%s'", finalizerName))
```

Comment:
This log line is the actual culprit of the issue. I moved it down a bit after some conditions of the action are verified. This should fix the issue.